src/golang/lib/workflow/scheduler/schedule.go

```diff
@@ -236,6 +237,31 @@ func ScheduleOperator(
 		)
 	}
 
+	if opSpec.IsSystemMetric() {
+		if len(outputContentPaths) != 1 {
+			return "", ErrWrongNumArtifactContentPaths
+		}
+		if len(outputMetadataPaths) != 1 {
+			return "", ErrWrongNumArtifactMetadataPaths
+		}
+		// We currently allow the spec to contain multiple input_metadata paths.
+		// A system metric currently spans over a single operator, so we enforce that here
```

Comment:
I would also like to see this comment on the `input_metadata_path` field on the spec struct declaration as well, since that's where most people would first find it.